### DISCUSSION TITLE
Can I use a combobox control in a column of Datagrid?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#14103] [Q&A]
Hi, is it possible?

**Comment by thevortexcloud** [comment]
You can use any control in a datagrid column with a template column. It lets you put completely arbitrary controls/layouts in it.

**Comment by timunie** [comment]
https://docs.avaloniaui.net/docs/reference/controls/datagrid/datagridcolumns

**Reply by IronEaterBeast** [reply]
Unable to find suitable setter or adder for property Items of type Avalonia.Controls:Avalonia.Controls.ItemsControl for argument Avalonia.Markup.Xaml:Avalonia.Markup.Xaml.MarkupExtensions.CompiledBindingExtension, available setter parameter lists are:
System.Object	

When my code is:
`<DataGrid.Columns>

				<DataGridTemplateColumn Header="My Column">

					<DataGridTemplateColumn.CellTemplate>

						<DataTemplate>

							<ComboBox Items="{Binding MyItems2}"/>

						</DataTemplate>

					</DataGridTemplateColumn.CellTemplate>

				</DataGridTemplateColumn>

</DataGrid.Columns>`


and C# code is:
`public class WpPbnModel

{

    public List<string> MyItems2 { get; set; }

}`

**Reply by thevortexcloud** [reply]
I am pretty sure it's `ItemSource` not `Items`

**Reply by IronEaterBeast** [reply]
> I am pretty sure it's `ItemSource` not `Items`

Yeah, I just found that, thank you. Do you know how to get the selected item in c# if there are many lines in the Datagrid?

**Reply by thevortexcloud** [reply]
Typically you bind to a selected item property on your view model. But if you are not using MVVM I am not sure. You can probably get to the combo box by iterating through the grid somehow.

**Reply by timunie** [reply]
Better to follow MVVM pattern here, really.
